newrelic-alerts-configurator/src/main/java/com/ocadotechnology/newrelic/alertsconfigurator/configuration/PolicyConfiguration.java

```diff
@@ -23,35 +27,211 @@
  * </ul>
  */
 @Getter
-@Builder
 public class PolicyConfiguration {
     /**
      * Your policy name
      */
-    @NonNull
     private String policyName;
     /**
      * Rollup strategy options for your alerts policy
      */
-    @NonNull
     private IncidentPreference incidentPreference;
     /**
      * Collection of {@link Channel} configurations
      */
-    @Singular
-    private Collection<Channel> channels = Collections.emptyList();
+    private Collection<Channel> channels;
     /**
      * Collection of {@link Condition} configurations
      */
-    @Singular
-    private Collection<Condition> conditions = Collections.emptyList();
+    private Collection<Condition> conditions;
     /**
      * Collection of {@link ExternalServiceCondition} configurations
      */
-    @Singular
-    private Collection<ExternalServiceCondition> externalServiceConditions = Collections.emptyList();
+    private Collection<ExternalServiceCondition> externalServiceConditions;
+    /**
+     * Collection of {@link NrqlCondition} configurations
+     */
+    private Collection<NrqlCondition> nrqlConditions;
+
+    @ConstructorProperties(
+            {"policyName", "incidentPreference", "channels", "conditions", "externalServiceConditions", "nrqlConditions"})
+    PolicyConfiguration(String policyName,
+                        IncidentPreference incidentPreference,
+                        Collection<Channel> channels, Collection<Condition> conditions,
+                        Collection<ExternalServiceCondition> externalServiceConditions,
+                        Collection<NrqlCondition> nrqlConditions) {
+        if (policyName == null) {
+            throw new NullPointerException("policyName");
+        }
+        if (incidentPreference == null) {
+            throw new NullPointerException("incidentPreference");
+        }
+
+        this.policyName = policyName;
+        this.incidentPreference = incidentPreference;
+        this.channels = channels;
+        this.conditions = conditions;
+        this.externalServiceConditions = externalServiceConditions;
+        this.nrqlConditions = nrqlConditions;
+    }
+
+    public static PolicyConfigurationBuilder builder() {
+        return new PolicyConfigurationBuilder();
+    }
 
     public enum IncidentPreference {
         PER_POLICY, PER_CONDITION, PER_CONDITION_AND_TARGET
     }
+
+    public static class PolicyConfigurationBuilder {
```

Comment:
This builder class is delomboked `@Builder` with only one change - it allows nullable condition collections